### ISSUE TITLE
JWT_SESSION_FIELD and subscriptions

### ISSUE THREAD

**Issue by OverFlow636** [Accenture/reactive-interaction-gateway#164]
Setting `JWT_SESSION_FIELD` for example to /username causes requests to `/_rig/v1/connection/ws/xxx/subscriptions` to 500

In the logs the crash happens for this
```
Request: PUT /_rig/v1/connection/ws/xxx/subscriptions
** (exit) an exception was raised:
    ** (Protocol.UndefinedError) protocol Enumerable not implemented for {:ok, %{"email" => "test@test.com", "exp" => 4103258143, "fullname" => "TEST FULL", "iat" => 4103258143, "name" => "John Doe", "sub" => "1234567890", "username" => "testusername"}}. This protocol is implemented for: Date.Range, File.Stream, Function, GenEvent.Stream, HashDict, HashSet, IO.Stream, List, Map, MapSet, Range, Stream, Timex.Interval
        (elixir) /usr/local/src/elixir/lib/elixir/lib/enum.ex:1: Enumerable.impl_for!/1
        (elixir) /usr/local/src/elixir/lib/elixir/lib/enum.ex:141: Enumerable.reduce/3
        (elixir) lib/enum.ex:2979: Enum.reduce/3
        (elixir) lib/enum.ex:1925: Enum."-reduce/3-lists^foldl/2-0-"/3
        (rig_auth) lib/rig_auth/session.ex:43: RigAuth.Session.do_update/3
        (rig_inbound_gateway) lib/rig_inbound_gateway_web/v1/subscription_controller.ex:72: RigInboundGatewayWeb.V1.SubscriptionController.set_subscriptions/2
        (rig_inbound_gateway) lib/rig_inbound_gateway_web/v1/subscription_controller.ex:1: RigInboundGatewayWeb.V1.SubscriptionController.action/2
        (rig_inbound_gateway) lib/rig_inbound_gateway_web/v1/subscription_controller.ex:1: RigInboundGatewayWeb.V1.SubscriptionController.phoenix_controller_pipeline/2
pid=<0.2328.0>
```

I have tried different fields inside the jwt, and using an extractor to try and get the field, but nothing has worked.

Im using `docker run -p 4000:4000 -p 4010:4010 -e SUBSCRIPTION_CHECK=jwt_validation -e JWT_SESSION_FIELD=username -e JWT_SECRET_KEY=secrettoken -e HTTPS_CERTFILE=cert/selfsigned.pem -e HTTPS_KEYFILE=cert/selfsigned_key.pem accenture/reactive-interaction-gateway:2.0.2` to start the service

I am able to call subscriptions successfully without the `JWT_SESSION_FIELD`, but as soon as i add it, they start crashing.

**Comment by kevinbader** [comment]
Thanks for reporting - I'll take a look at this as soon as possible.

**Comment by kevinbader** [comment]
I've prepared a fix for this, but I'm waiting for #166 to land in master first. The fix will most likely get published with the next (bugfix) release within the next two weeks.

Regarding your example, please make sure to pass a JSON pointer rather than a field name - from the docs:

> The JWT_SESSION_FIELD is specified using the JSON Pointer notation. Given that the JWT contains a user ID in its "userId" field, the configuration could look like this: `JWT_SESSION_FIELD=/userId`.

**Comment by kevinbader** [comment]
@OverFlow636 should be fixed with the latest Docker image. Can you confirm?

**Comment by OverFlow636** [comment]
@kevinbader when trying the latest update i ran into an issue where the old environment variables i was using to start the service are now crashing it (https certfile and keyfile)

When i run the following it crashes, however removing the HTTPS env vars allows it to run now
`docker run -p 4000:4000 -p 4010:4010 -e SUBSCRIPTION_CHECK=jwt_validation -e JWT_SESSION_FIELD=/username -e JWT_SECRET_KEY= secrettoken -e HTTPS_CERTFILE=cert/selfsigned.pem -e HTTPS_KEYFILE=cert/selfsigd_key.pem accenture/reactive-interaction-gateway:latest`

With the modified command it will now accept subscriptions, however its no longer validating jwt's
`docker run -p 4000:4000 -p 4010:4010 -e SUBSCRIPTION_CHECK=jwt_validation -e JWT_SESSION_FIELD=/username -e JWT_SECRET_KEY=secrettoken accenture/reactive-interaction-gateway:latest`
and testing with https://github.com/Accenture/reactive-interaction-gateway/blob/master/examples/ws-demo-jwt.html

the call to subscription 204s when it should error out because the jwt in the example file is not hashed with the correct `JWT_SECRET_KEY`

when i was using 2.0.2 requests to subscribe would 500 if their jwts were invalid.

**Comment by kevinbader** [comment]
Hm I see... will fix this and let you know. Thanks!

**Comment by kevinbader** [comment]
Should be fixed with #186 and included in the upcoming release.
